app/org/sagebionetworks/bridge/dynamodb/DynamoMpowerVisualizationDao.java

```diff
@@ -0,0 +1,55 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import java.util.List;
+import javax.annotation.Resource;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.joda.time.LocalDate;
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.dao.MpowerVisualizationDao;
+import org.sagebionetworks.bridge.json.BridgeObjectMapper;
+
+/** DDB implementation of mPower visualization. */
+@Component
+public class DynamoMpowerVisualizationDao implements MpowerVisualizationDao {
+    private DynamoDBMapper mapper;
+
+    /** DDB mapper for the mPower visualization table. */
+    @Resource(name = "mpowerVisualizationDdbMapper")
+    final void setDdbMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public JsonNode getVisualization(String healthCode, LocalDate startDate, LocalDate endDate) {
+        // hash key comes from study and health code
+        DynamoMpowerVisualization hashKey = new DynamoMpowerVisualization();
```

Comment:
Discussed this on IM. The code isn't specific to mPower, but the feature definitely is. We can't know for sure whether this will fit other use cases until we have other use cases, and given the timeline and constraints, we don't have time to design this in a generic way. As such, I'd prefer clearly labeling this as mPower-specific until we know it's not.

We could always rename it later.